src/network.rs

```diff
@@ -157,20 +164,29 @@ fn listen_stream(
     });
 }
 
-fn connect_event_stream(url: &Url, stream: &StreamWrapper, last_event_id: &LastIdWrapper) -> Result<TcpStream, Error> {
+fn connect_event_stream(url: &Url, last_event_id: &LastIdWrapper) -> Result<InnerStream, Error> {
     let connection_stream = event_stream_handshake(url, last_event_id)?;
-    let mut stream = stream.lock().unwrap();
-    *stream = Some(connection_stream.try_clone().unwrap());
```

Comment:
I believe this block is important, as the close method uses this reference to close the connection. If you are able to add it back it should fix the issue.